web_translate_dialog/static/src/js/web_translate_dialog.js

```diff
@@ -121,36 +117,36 @@ var translateDialog = Dialog.extend({
             deferred = [];
 
         this.$el.find('.oe_translation_field').val('').removeClass('touched');
-        _.each(self.languages, function(lg) {
-            var context = new Context(session.user_context, {lang: lg.code});
-            var deff = $.Deferred();
-            deferred.push(deff);
-            rpc.query({
-            model: self.view.modelName,
-            method: 'read',
+
+        //var context = new Context(session.user_context, {lang: lg.code});
+        var deff = $.Deferred();
+        deferred.push(deff);
+        rpc.query({
+            model: this.view.modelName,
+            method: 'get_field_translations',
             args: [
-                [self.res_id],
+                [this.res_id],
             ],
             kwargs: {
-                fields: self.translatable_fields,
-                context: context.eval(),
+                field_name: this.translatable_field,
             },
-            }).done(
-                    function (rows) {
-                        
-                        if (rows[0]){
-                            self.set_fields_values(lg, rows[0]);
-                            deff.resolve();
-                        }
+        }).done(
+            function (res) {
+                if (res[self.res_id]){
+                    _.each(res[self.res_id], function(translation, lang) {
+                        self.set_fields_values(lang, translation[1]);
                     });
-        });
+                    deff.resolve();
+                }
+            });
+
         return deferred;
     },
     on_button_save: function() {
         var translations = {},
             self = this,
             save_mutex = new Mutex();
-        self.$el.find('.oe_translation_field.touched').each(function() {
+        this.$el.find('.oe_translation_field.touched').each(function() {
```

Comment:
uneeded change

Reply:
Isn't it 'better' to use `this` if it's available ?

Reply:
I would say so too, here the use of self is not needed `self` AFAIK